tests/Unit/IO/Test_VolumeData.py

```diff
@@ -42,26 +105,22 @@ def test_insert_vol(self):
 
     # Test the header was generated correctly
     def test_vol_get_header(self):
-        h5_file = spectre_h5.H5File(file_name=self.file_name_w,
+        os.remove(self.file_name)
+        h5_file = spectre_h5.H5File(file_name=self.file_name,
                                     append_to_file=True)
         h5_file.insert_vol(path="/element_data", version=0)
         vol_file = h5_file.get_vol(path="/element_data")
         self.assertEqual(vol_file.get_header()[0:20], "#\n# File created on ")
         h5_file.close()
 
-    # The other tests require external data, so they use the included h5 file
-    # `VolTestData.h5` which contains spectre output data (see above).
-    # Test the observation ids and values are correctly retrived
+    # Using volume data H5 file to do rest of the tests
+    # Test that observation ids and values are retrieved correctly
```

Comment:
I suggest you split the class here so that you have two classes: `TestVolumeDataWriting` and `TestVolumeData`, for example. Then you can move the `setUp` and `tearDown` code to the `TestVolumeData` class, so that the `TestVolumeDataWriting` class doesn't need to write an H5 file to disk just to delete it again when running the `test_insert_vol` and `test_vol_get_header` functions.

Reply:
Yes, this sounds good